scripts/cherry-pick

```diff
@@ -92,35 +101,32 @@ run_quilt() {
         quilt --quiltrc - "$@"
 }
 
-main() {
-    local short_opts="ho:v"
-    local long_opts="help,verbose"
-    local getopt_out=""
-    getopt_out=$(getopt --name "${0##*/}" \
-        --options "${short_opts}" --long "${long_opts}" -- "$@") &&
-        eval set -- "${getopt_out}" ||
-        { bad_Usage; return; }
-
-    local cur="" next=""
-
-    while [ $# -ne 0 ]; do
-        cur="$1"; next="$2";
-        case "$cur" in
-            -h|--help) Usage ; exit 0;;
-            -v|--verbose) VERBOSITY=$((${VERBOSITY}+1));;
-            --) shift; break;;
-        esac
-        shift;
-    done
-
-	[ -n "$TEMP_D" ] ||
-        TEMP_D=$(mktemp -d "${TMPDIR:-/tmp}/${0##*/}.XXXXXX") ||
-		{ error "failed to make tempdir"; return 1; }
-	trap cleanup EXIT
+NEW_VERSION=""
 
-    [ $# -gt 0 ] || { bad_Usage "must provide commit-ish"; return; }
+drop_cherry_pick_from_daily_branch() {
+    local local_cpick_commit="$1" daily_remote="$2"
+    cur_br=$(git rev-parse --abbrev-ref HEAD) ||
+        { error "failed to get current branch name"; return 1; }
+    daily_br=${cur_br/ubuntu/ubuntu\/daily}
+    git fetch ${daily_remote}
+    echo "Backing out cherry-pick from ${daily_br} branch"
+    if [ $(git show-ref refs/heads/${daily_br}) ]; then
+        # Branch already exists so refresh it to remote/ubuntu/daily/<series>
+        git checkout ${daily_br}
```

Comment:
We have a lot of git commands.  we should use a git wrapper which checks the error code for us:

```
!/bin/bash

error() { echo "$@" 1>&2; }
fail() { [ $# -eq 0 ] || error "$@"; exit 1; }

gitcmd() {
    git $@ || fail "Failed running git $@"
}

gitcmd branchname
gitcmd checkout foobar-wark
```

```
(crispyboi) cloud-init % bash -x gw.sh 
+ gitcmd branchname
+ git branchname
master
+ gitcmd checkout foobar-wark
+ git checkout foobar-wark
error: pathspec 'foobar-wark' did not match any file(s) known to git
+ fail 'Failed running git checkout' foobar-wark
+ '[' 2 -eq 0 ']'
+ error 'Failed running git checkout' foobar-wark
+ echo 'Failed running git checkout' foobar-wark
Failed running git checkout foobar-wark
+ exit 1
```